tests/FSharp.Compiler.ComponentTests/Signatures/NestedTypeTests.fs

```diff
@@ -0,0 +1,100 @@
+﻿module FSharp.Compiler.ComponentTests.Signatures.NestedTypeTests
+
+open Xunit
+open FsUnit
+open FSharp.Test.Compiler
+open FSharp.Compiler.ComponentTests.Signatures.TestHelpers
+
+[<Fact>]
+let ``Nested type with generics`` () =
+    let CSLib =
+        CSharp """
+namespace Lib
+{
+    public class Upper<T>
+    {
+        public class Lower<U>
+        {
+            public void Meh()
+            {
+                
+            }
+        }
+    }
+}
+"""
+
+    FSharp
+        """
+module Sample
+
+open Lib
+
+let f (g: Upper<int>.Lower<string>) = g.Meh()
+"""
+    |> withReferences [ CSLib ]
+    |> printSignatures
+    |> should equal
+        """
+module Sample
+
+val f: g: Lib.Upper<int>.Lower<string> -> unit"""
+
+[<Fact>]
+let ``Multiple generics in nested type`` () =
+    let CSLib =
+        CSharp """
+namespace Lib
+{
+    public class Root<A, B, C, D, E>
+    {
+        public class Foo<T, U, V, W>
+        {
+            public class Bar<X, Y, Z>
+            {
+                public void Meh()
+                {
+
+                }
+            }
+        }
+    }
+}
+"""
+
+    FSharp
+        """
+module Sample
+
+open System
+open Lib
+
+let f (g: Root<TimeSpan,TimeSpan,TimeSpan,TimeSpan,TimeSpan>.Foo<int, float, string, System.DateTime>.Bar<char, int, string>) = g.Meh()
+"""
+    |> withReferences [ CSLib ]
+    |> printSignatures
+    |> should equal
+        """
+module Sample
+
+val f: g: Lib.Root<System.TimeSpan,System.TimeSpan,System.TimeSpan,System.TimeSpan,System.TimeSpan>.Foo<int,float,string,System.DateTime>.Bar<char,int,string> -> unit"""
+
+#if NETCOREAPP
```

Comment:
You can use `[<FactForNETCOREAPP>]`, it was created for this purpose.

(it will also allow us to see such tests at a glance once we want to have an overview of platform-specific features)

Reply:
Thanks